syntax = "proto3";

package tweety;

message Message {
  string message_id = 1;
  string user_id = 2;
  string content = 3;
  string timestamp = 4; // ISO 8601 format
  bool is_repost = 5; // Indicates whether the message is a repost
  string original_message_id = 6; // Original poster's username, if a repost
}

message User {
    string user_id = 1;
    string password_hash = 2;
}

message UserFollows {
    repeated string followed_user_ids = 1;
}

message UserMessages {
    repeated string message_ids = 1;
}
